src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternGenerator.cpp

```diff
@@ -0,0 +1,91 @@
+
+
+#include <OpenMS/CHEMISTRY/Element.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopePatternGenerator.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+#include <cmath>
+#include <fstream>
+
+using namespace std;
+
+namespace OpenMS
+{
+  IsotopePatternGenerator::IsotopePatternGenerator() : IsotopeDistribution()
+  {}
+
+  IsotopePatternGenerator::IsotopePatternGenerator(const IsotopeDistribution& rhs) :
+    IsotopeDistribution(rhs)
+  {
+    
+  }
+  void IsotopePatternGenerator::merge(ContainerType& raw, double resolution)
+  {
+    //raw must be ordered to work correctly ascending order on power field
+
+    UInt output_size = ceil((raw.back().getMZ() - raw.front().getMZ())/resolution);
+    LOG_INFO << "output size " << output_size << endl;
+    LOG_INFO << "raw size " << raw.size() <<endl;
+
+    distribution_.clear();
+    distribution_.resize(output_size, Peak1D(0, 0));
+
+    for(auto& p : raw)
```

Comment:
nitpick: better to use `const auto&` to make clear that no modifications are made